lib/puppet/provider/archive/curl.rb

```diff
@@ -7,8 +7,14 @@
 
   def curl_params(params)
     if resource[:username]
-      create_netrcfile
-      params += ['--netrc-file', @netrc_file.path]
+      if resource[:username] =~ %r{\s} || resource[:password] =~ %r{\s}
+        Puppet.warning('Username or password contains a space.  Unable to use netrc file to hide credentials')
+        account = [resource[:username], resource[:password]].compact.join(':')
```

Comment:
This is the implementation from before https://github.com/voxpupuli/puppet-archive/pull/399